syntax = "proto3";

///////////////////////////////////////////////////////////////////////////////
///////////////////////////////////////////////////////////////////////////////

package Calculator;

///////////////////////////////////////////////////////////////////////////////
///////////////////////////////////////////////////////////////////////////////

message Pair {
    int32 lhs = 1;
    int32 rhs = 2;
}

message AddResult {
    int32 value = 1;
}
message SubResult {
    int32 value = 1;
}
message MulResult {
    int32 value = 1;
}
message DivResult {
    int32 value = 1;
}

service Service {
    rpc add(Pair) returns(AddResult);
    rpc sub(Pair) returns(SubResult);
    rpc mul(Pair) returns(MulResult);
    rpc div(Pair) returns(DivResult);
}

///////////////////////////////////////////////////////////////////////////////
///////////////////////////////////////////////////////////////////////////////
